plugin/completion.py

```diff
@@ -105,6 +113,44 @@ def get_text_edit_range(text_edit: Union[TextEdit, InsertReplaceEdit]) -> Range:
     return text_edit['range']
 
 
+def is_range(val: Any) -> TypeGuard[Range]:
+    return isinstance(val, dict) and 'start' in val and 'end' in val
+
+
+def is_edit_range(val: Any) -> TypeGuard[CompletionEditRange]:
+    return isinstance(val, dict) and 'insert' in val and 'replace' in val
+
+
+def completion_with_defaults(item: CompletionItem, item_defaults: CompletionItemDefaults) -> CompletionItem:
+    """ Currently supports defaults for: ["editRange", "insertTextFormat", "data"] """
+    default_text_edit = None  # type: Optional[Union[TextEdit, InsertReplaceEdit]]
```

Comment:
```suggestion
    """ Currently supports defaults for: ["editRange", "insertTextFormat", "data"] """
    if not item_defaults:
        return item
    default_text_edit = None  # type: Optional[Union[TextEdit, InsertReplaceEdit]]
```

All of the following dict accesses and if-checks are unnecessary if the server doesn't provide the CompletionList.itemDefaults, which is probably the case for most language servers currently.